includes/payment-methods/class-wechatpay-payment-method.php

```diff
@@ -106,53 +106,38 @@ public function get_currencies() {
 		// Map countries to their primary currencies.
 		switch ( $account_country ) {
 			case Country_Code::AUSTRALIA:
-				return [ Currency_Code::AUSTRALIAN_DOLLAR, Currency_Code::CHINESE_YUAN ];
+				return [ Currency_Code::AUSTRALIAN_DOLLAR ];
 			case Country_Code::CANADA:
-				return [ Currency_Code::CANADIAN_DOLLAR, Currency_Code::CHINESE_YUAN ];
-			case Country_Code::CHINA:
-				return [ Currency_Code::CHINESE_YUAN ];
-			case Country_Code::DENMARK:
-				return [ Currency_Code::DANISH_KRONE, Currency_Code::EURO, Currency_Code::CHINESE_YUAN ];
+				return [ Currency_Code::CANADIAN_DOLLAR ];
 			case Country_Code::HONG_KONG:
-				return [ Currency_Code::HONG_KONG_DOLLAR, Currency_Code::CHINESE_YUAN ];
-			case Country_Code::JAPAN:
-				return [ Currency_Code::JAPANESE_YEN, Currency_Code::CHINESE_YUAN ];
-			case Country_Code::NORWAY:
-				return [ Currency_Code::NORWEGIAN_KRONE, Currency_Code::EURO, Currency_Code::CHINESE_YUAN ];
+				return [ Currency_Code::HONG_KONG_DOLLAR ];
 			case Country_Code::SINGAPORE:
-				return [ Currency_Code::SINGAPORE_DOLLAR, Currency_Code::CHINESE_YUAN ];
-			case Country_Code::SWEDEN:
-				return [ Currency_Code::SWEDISH_KRONA, Currency_Code::EURO, Currency_Code::CHINESE_YUAN ];
-			case Country_Code::SWITZERLAND:
-				return [ Currency_Code::SWISS_FRANC, Currency_Code::EURO, Currency_Code::CHINESE_YUAN ];
+				return [ Currency_Code::SINGAPORE_DOLLAR ];
 			case Country_Code::UNITED_KINGDOM:
-				return [ Currency_Code::POUND_STERLING, Currency_Code::CHINESE_YUAN ];
+				return [ Currency_Code::POUND_STERLING ];
 			case Country_Code::UNITED_STATES:
-				return [ Currency_Code::UNITED_STATES_DOLLAR, Currency_Code::CHINESE_YUAN ];
+				return [ Currency_Code::UNITED_STATES_DOLLAR ];
 			default:
-				// For all European countries in the supported list, return EUR and CNY.
+				// For all European countries in the supported list, return EUR.
 				if ( in_array(
 					$account_country,
 					[
 						Country_Code::AUSTRIA,
 						Country_Code::BELGIUM,
-						Country_Code::FINLAND,
 						Country_Code::FRANCE,
 						Country_Code::GERMANY,
 						Country_Code::IRELAND,
 						Country_Code::ITALY,
-						Country_Code::LUXEMBOURG,
 						Country_Code::NETHERLANDS,
-						Country_Code::PORTUGAL,
 						Country_Code::SPAIN,
 					],
 					true
 				) ) {
-					return [ Currency_Code::EURO, Currency_Code::CHINESE_YUAN ];
+					return [ Currency_Code::EURO ];
 				}
 
-				// Default to Chinese Yuan.
-				return [ Currency_Code::CHINESE_YUAN ];
+				// Defaulted to unsupported currency.
+				return [ 'UNSUPPORTED' ];
```

Comment:
I'm not sure if this is the best way to handle the default, but I guess it would make sense follow the same standard in each payment method.

Reply:
We decided to keep things as they are, so feel free to ignore this comment.